networkx/generators/tests/test_directed.py

```diff
@@ -105,11 +126,14 @@ def test_regularity(self):
         G = random_uniform_k_out_graph(n, k, seed=42)
         assert all(d == k for v, d in G.out_degree())
 
-    def test_no_self_loops(self):
+    @pytest.mark.parametrize("with_replacement", [False, True])
+    def test_no_self_loops(self, with_replacement):
         """Tests for forbidding self-loops."""
         n = 10
         k = 3
-        G = random_uniform_k_out_graph(n, k, self_loops=False)
+        G = random_uniform_k_out_graph(
```

Comment:
Nice work parametrizing this - I think it's nicer now. There's still one component that I think would improve this test: adding a check for the output graph type, which will change depending on the `with_replacement` value. Can you think of a way to do so in the parametrized test?

Reply:
Yes, I can. Thanks for the review

Reply:
This is one way to do it; another way would be to add the expected result to the parametrization. This is arguably a better pattern as the expected results stay tightly coupled to the inputs, and there is no conditional branching within the test itself. The end result might look something like:

```python
@pytest.mark.parametrize(
    ("with_replacement", "expected_type"),
    (
        (True, nx.MultiDiGraph),  # with_replacement=True results in a MultiDiGraph
        (False, nx.DiGraph),
    ),
)
def test_no_self_loops(with_replacement, expected_type):
    G = random_uniform_k_out_graph(
        10, 3, self_loops=False, with_replacement=with_replacement
    )
    assert isinstance(G, expected_type)
    assert nx.number_of_selfloops(G) == 0
    assert all(d == k for _, d in G.out_degree())
```